taxi/forms.py

```
@@ -0,0 +1,45 @@
+from django import forms
+from django.contrib.auth import get_user_model
+from django.contrib.auth.forms import UserCreationForm
+from django.core.exceptions import ValidationError
+
+from taxi.models import Car
+
+
+class DriverLicenseUpdateForm(forms.ModelForm):
+
+    class Meta:
+        model = get_user_model()
+        fields = UserCreationForm.Meta.fields + ("license_number",)
+
+    def clean_license_number(self):
+        license_number = self.cleaned_data["license_number"]
+
+        if len(license_number) < 8:
+            return ValidationError(
+                "License number should be at least 8 characters long"
+            )
+
+        if license_number[:3] != license_number[:3].upper():
+            return ValidationError(
+                "First 3 characters should be uppercase letters"
+            )
+
+        if not license_number[-5:].isdigit():
+            return ValidationError(
+                "Last 5 characters should be numbers"
+            )
+
+        return license_number
```

The validation logic in this method has a few issues:

1.  You must `raise ValidationError` instead of returning it. Returning the error does not stop the form validation process.
2.  The length check on line 18 is for 'at least 8 characters', but the requirement specifies **exactly** 8 characters.
3.  The check on line 23 only verifies if the first 3 characters are uppercase, but not if they are letters. For example, '123' would pass this check. You need to also check if they are alphabetic characters.